certbot/certbot/util.py

```diff
@@ -48,6 +50,200 @@ class CSR(NamedTuple):
+                    return self.Comparison.LESS
+
+        if len(self.version_components) > len(other.version_components):
+            for version_component in self.version_components[len(other.version_components)::]:
+                if version_component != 0:
+                    return self.Comparison.GREATER
+        elif len(self.version_components) < len(other.version_components):
+            for version_component in other.version_components[len(self.version_components)::]:
+                if version_component != 0:
+                    return self.Comparison.LESS
+
+        return self.Comparison.EQUAL
+
+    def try_risky_comparison(self, other: Any) -> int:
+        """Compares the LooseVersion to another value.
+
+        If the other value is another LooseVersion, the version components are compared. Otherwise,
+        an exception is raised.
+
+        Comparison is performed element-wise. If the version components being compared are of
+        different types, the two versions are considered incomparible. Otherwise, if either of the
+        components is not equal to the other, less or greater is returned based on the comparison's
+        result. In case the two versions are of different lengths, some elements in the longer
+        version have not yet been compared. If these are all equal to zero, the two versions are
+        equal. Otherwise, the longer version is greater.
+
+        If the two versions are incomparible, an exception is raised. Otherwise, the returned
+        integer indicates the result of the comparison. If self == other, 0 is returned.
+        If self > other, 1 is returned. If self < other -1 is returned.
+
+        Examples:
+        Equality:
+        - LooseVersion('1.0').try_risky_comparison(LooseVersion('1.0')) -> 0
+        - LooseVersion('2.0.0a').try_risky_comparison(LooseVersion('2.0.0a')) -> 0
+        Inequality:
+        - LooseVersion('2.0.0').try_risky_comparison(LooseVersion('1.0')) -> 1
+        - LooseVersion('1.0.1').try_risky_comparison(LooseVersion('2.0a')) -> -1
+        Incomparability:
+        - LooseVersion('1a').try_risky_comparison(LooseVersion('1.0')) -> ValueError
+        """
+        if not isinstance(other, type(self)):
+            raise TypeError(f'Comparison not supported between LooseVersion and {type(other)}')
+
+        comparison = self.__compare(other)
+
+        if comparison == self.Comparison.EQUAL:
+            return 0
+        elif comparison == self.Comparison.LESS:
+            return -1
+        elif comparison == self.Comparison.GREATER:
+            return 1
+        else:
+            raise ValueError("Cannot meaningfully compare LooseVersion {} with LooseVersion {} "
+                             "due to comparison of version components with different types."
+                             .format(self.version_components, other.version_components))
+
+    # Provide helper methods for comparison without having to use the returned integer value.
```

Comment:
I don't think we should provide these. While it's not common in modern Python, a function like this is fairly common in other languages. See https://en.wikipedia.org/wiki/Three-way_comparison. You can check for what you want by using the `<=`, `<`, `>`, and `>=` operators with the return value of the function and `0`. See https://github.com/python/cpython/blob/579e985e75fd682e314c7b0cce6b1751abe8a7d0/Lib/functools.py#L212-L221.